
Betreff:
[PATCH] tethering: Cleanup tethering when connman is shutting down
Von:
Jukka Rissanen <jukka.rissanen@linux.intel.com>
Datum:
26.06.2013 09:40
An:
connman@connman.net

We do not want to leave the tether interface and bridge hanging
around after shutdown.
---
 src/tethering.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/src/tethering.c b/src/tethering.c
index 0b373b9..03ed02a 100644
--- a/src/tethering.c
+++ b/src/tethering.c
@@ -532,10 +532,10 @@ int __connman_tethering_init(void)
 
 void __connman_tethering_cleanup(void)
 {
-	DBG("");
+	DBG("enabled %d", tethering_enabled);
 
 	__sync_synchronize();
-	if (tethering_enabled == 0) {
+	if (tethering_enabled > 0) {
 		if (tethering_dhcp_server)
 			dhcp_server_stop(tethering_dhcp_server);
 		__connman_bridge_disable(BRIDGE_NAME);
--
1.7.11.7

